### PR TITLE
Add support for a profile specific kubeconfig file.

### LINKED ISSUE #7202: Standalone Terminal / Terminal Locking For K8s on Different Teleport Clusters
### What
Ability to have each terminal tab/window be treated as a standalone environment from a kubernetes context perspective. Some customers require the ability to work in multiple, separate teleport clusters, each with their own kubernetes cluster, which is currently not possible as each new tsh login overrides the previous kube context even if `eval $(tsh env)` is used. 

### Why
Customer ask. Having the ability to treat each terminal tab/window as a standalone environment for kube contexts would also help prevent accidental context switching as devs/users could pin kube contexts to a single terminal tab/window without having to log in/out between contexts each time they switched clusters.

### Workaround
n/a

### PR DESCRIPTION
**Pull request by Joerger**
When a user logs in to a kube cluster with `tsh login --kube-cluster=kube01` or `tsh kube login kube02`, in addition to updating their `~/.kube/config` or `$KUBECONFIG`, a standalone kubeconfig for the given cluster will be generated in `~/.tsh/keys/${PROXY}/${USER}-kube/${TELE_CLUSTER}/kube01-kubeconfig`. This kubeconfig has a single cluster (kube01) which is set as the context.

After logging in to a kube cluster, a user can then run `tsh env` to get and run `export KUBECONFIG=~/.tsh/.../kube01-kubeconfig`.

Example: 
```bash
> tsh kube ls
Kube Cluster Name Selected 
----------------- -------- 
kube01            *        
kube02   

> eval $(tsh env)

> echo $KUBECONFIG
~/.tsh/keys/proxy.example.com/dev-kube/example.com/kube01-kubeconfig

> tsh kube login kube02
ERROR: profile specific kubeconfig is in use, run 'eval $(tsh env --unset)' to switch contexts to another kube cluster

#### Change windows/tabs

> tsh kube login kube02
Logged into kubernetes cluster "kube02"

> tsh kube ls
Kube Cluster Name Selected 
----------------- -------- 
kube01                     
kube02            *        

#### Change back to the first window

> tsh kube ls
Kube Cluster Name Selected 
----------------- -------- 
kube01            *        
kube02      
```

Closes https://github.com/gravitational/teleport/issues/7202

### CODE REVIEW

**Comment by russjones** [comment]
@Joerger We have a `unset` subcommand as well: https://github.com/gravitational/teleport/blob/master/tool/tsh/tsh.go#L473

File: tool/tsh/kube.go [hunk 1]
```diff
@@ -360,7 +373,20 @@ func updateKubeConfig(cf *CLIConf, tc *client.TeleportClient) error {
...
-	return trace.Wrap(kubeconfig.Update("", *values))
+	if path == "" {
+		path = kubeconfig.PathFromEnv()
+	}
+
+	// If this is a profile specific kubeconfig, we only need
+	// to put the selected kube cluster into the kubeconfig.
+	if keypaths.IsProfileKubeConfigPath(path) {
+		if !strings.Contains(path, cf.KubernetesCluster) {
+			return trace.BadParameter("profile specific kubeconfig is in use, unset $KUBECONFIG to switch contexts to another kube cluster")
```
Comment (russjones) [inline tool/tsh/kube.go:384]: Why not just tell the user to eval `tsh unset`?

File: api/utils/keypaths/keypaths.go [hunk 2]
```diff
@@ -226,16 +228,32 @@ func KubeCertPath(baseDir, proxy, username, cluster, kubename string) string {
 	return filepath.Join(KubeCertDir(baseDir, proxy, username, cluster), kubename+fileExtTLSCert)
```
Comment (andrejtokarcik) [inline api/utils/keypaths/keypaths.go:228]: I noticed some of the `Database*` functions defined above have comments referring to "kube directory". That's probably a mistake, would you mind fixing the comments as part of this PR?

File: api/utils/keypaths/keypaths.go [hunk 3]
```diff
@@ -45,6 +45,8 @@ const (
 	dbDirSuffix = "-db"
 	// kubeDirSuffix is the suffix of a sub-directory where kube TLS certs are stored.
 	kubeDirSuffix = "-kube"
+	// kubeConfigSuffix is the suffix of a kubeconfig file stored under the keys directory.
+	kubeConfigSuffix = "-kubeconfig"
 )
 
 // Here's the file layout of all these keypaths.
```
Comment (andrejtokarcik) [inline api/utils/keypaths/keypaths.go:52]: Could you update the layout below to include a kubeconfig file?

**Review by andrejtokarcik — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Comment by Joerger** [comment]
@klizhentas Please take a look when you can

**Review by klizhentas — APPROVED** [review]

**Pull request merged**